File: Algorithms/Searching/Linear Search/linear-search.cpp
/*
 * Linear Search
 *
 */

#include <iostream>
#include <cstdio>
#include <cmath>

using namespace std;

int linearSearch(int arr[], int n, int a) {
    for (int i = 0; i < n; ++i)
        if (arr[i] == a)
            return i;

    return -1;
}

int main(int argc, char const *argv[]) {
    int arr[] = {3,5,12,56,92,123,156,190,201,222};
    int a = 12;
    int n = sizeof(arr)/ sizeof(arr[0]);
    cout<<"Linear Search :: "<<linearSearch(arr, n, a)<<endl;
    return 0;
}

File: Algorithms/Searching/Linear Search/linear-search.py
"""
  Linear Search
"""


def LinearSearch(arr, k):
    for i in range(0, len(arr)):
        if(arr[i] == k):
            return i
    return -1


print(LinearSearch([3, 5, 12, 56, 92, 123, 156, 190, 201, 222], 12))
